### PR TITLE
Add controlnet + sdxl_base inference support

### PR DESCRIPTION
**Pull request by HaoyangLee**
TODO: 
- README
- adapt for demo/sampling.py (with streamlit)
- other control condition if needed, only support canny now

### CODE REVIEW

File: examples/stable_diffusion_xl/demo/sampling_without_streamlit.py [hunk 1]
```diff
@@ -36,6 +38,14 @@ def get_parser_sample():
     parser.add_argument("--negative_aesthetic_score", type=float, default=None)
     parser.add_argument("--sampler", type=str, default="EulerEDMSampler")
     parser.add_argument("--guider", type=str, default="VanillaCFG")
+    parser.add_argument(
+        "--guidance_scale",
+        type=float,
+        default=5.0,
+        help="unconditional guidance scale: "
+        "eps = eps(x, uncond) + scale * (eps(x, cond) - eps(x, uncond)). "
+        "Simplified: `uc + scale * (uc - prompt)`",
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/demo/sampling_without_streamlit.py:47]: `uc + scale * (uc - prompt)` -> For NoDynamicThresholding, `uncond + guidance_scale * (uncond - cond)`

File: examples/stable_diffusion_xl/cldm/cldm.py [hunk 2]
```diff
@@ -0,0 +1,375 @@
+import logging
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/cldm/cldm.py:1]: 是否将 `cldm` 移动到 `gm/cldm` 会更好

Comment (HaoyangLee) [reply examples/stable_diffusion_xl/cldm/cldm.py:1]: 目前对齐`stable_diffusion_v2/cldm`和`stable_diffusion_v2/ldm`的目录结构，便于后续把controlnet部分抽出来放到`examples/`目录下

File: examples/stable_diffusion_xl/demo/sampling_without_streamlit.py (hunk 1 above)
Comment (HaoyangLee) [reply examples/stable_diffusion_xl/demo/sampling_without_streamlit.py]: updated

File: examples/stable_diffusion_xl/scripts/run_infer_base_controlnet.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+export MS_PYNATIVE_GE=1
```
Comment (SamitHuang) [inline examples/stable_diffusion_xl/scripts/run_infer_base_controlnet.sh:1]: can remove it since it's hard-coded lately

Comment (HaoyangLee) [reply examples/stable_diffusion_xl/scripts/run_infer_base_controlnet.sh]: removed

**Review by zhanghuiyao — APPROVED** [review]

**Review by SamitHuang — APPROVED** [review]

**Pull request merged**